google/cloud/pubsub/internal/streaming_subscription_batch_source.cc

```diff
@@ -130,6 +131,7 @@ future<Status> StreamingSubscriptionBatchSource::AckMessage(
 
 future<Status> StreamingSubscriptionBatchSource::NackMessage(
     std::string const& ack_id) {
+  callback_->AddEvent(ack_id, "gl-cpp.nack_start");
```

Comment:
where are the tests that this class invokes some method on `callback_`?

We have a mock callback, we should be able to use that.

Reply:
> if we just have the unit testing in the callback_test.cc

Why? That doesn't sound right to me. The tracing batch callback unit test is where we should check that spans are made, have the right events. We still want something that gives us confidence that the interfaces are ever invoked. I don't think setting expectations will bloat the tests or distract the PRs.

The tests use the mock class already. You would just need to add expectations on what is called. Using a `NaggyMock<>` would reveal them.

https://github.com/googleapis/google-cloud-cpp/blob/c51990c5341c2ebf32ad370b3761bdcde216f6c3/google/cloud/pubsub/internal/streaming_subscription_batch_source_test.cc#L266-L267

Reply:
Ack, added the mock expect calls.